### PR TITLE
Update `Pod` documentation in the face of field reordering

### PR DESCRIPTION
**Pull request by fitzgen**
r? @nrc

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -89,9 +89,8 @@ pub fn read_strs_to_null<'a>(input: &'a [u8]) -> StrReaderIterator<'a> {
 /// * primitive numeric types (`u8`, `i64`, `f32`, etc.) are fine,
 /// * bools are fine, if the provided data ensures they may have only the values
 ///   `0` or `1` (note that this is a stricter requirement that C),
-/// * structs containing only `Pod` data are fine,
-/// * structs must be `repr(C)` or `repr(packed)`, if the former, the supplied
-///   data must have the correct alignment,
+/// * structs may be `Pod` if they have the `repr(C)` attribute to prevent field
```
Comment (nrc) [inline src/lib.rs]: I think using `repr(packed)` is still valid, since that also preserves ordering.

**Comment by nrc** [comment]
Thanks!

**Comment by fitzgen** [comment]
Oh! You're too fast! I added one more commit to also (re)document that all members must also be Pod

**Comment by fitzgen** [comment]
Its on the same branch. I made the commit via the web editor, so not sure how to open a new PR with it..

**Pull request merged**